### PR TITLE
Avoid throwing NullPointerExceptions manually.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: jspwiki-main/src/main/java/org/apache/wiki/providers/VersioningFileProvider.java [hunk 1]
```diff
@@ -685,11 +685,11 @@ private static class CachedProperties {
          */
         public CachedProperties( final String pageName, final Properties props, final long lastModified ) {
             if ( pageName == null ) {
-                throw new NullPointerException ( "pageName must not be null!" );
+                throw new IllegalArgumentException ( "pageName must not be null!" );
```
Comment (juanpablo-santos) [inline jspwiki-main/src/main/java/org/apache/wiki/providers/VersioningFileProvider.java:688]: I think this is one of those cases where `NullPointerException` really expresses the intent - the field must not be null

File: jspwiki-util/src/main/java/org/apache/wiki/util/TextUtil.java [hunk 2]
```diff
@@ -232,8 +232,11 @@ public static String replaceString( final String orig, final String src, final S
         if ( orig == null ) {
             return null;
         }
-        if ( src == null || dest == null ) {
-            throw new NullPointerException();
+        if ( src == null ) {
+            throw new IllegalArgumentException ( "src must not be null!" );
```
Comment (juanpablo-santos) [inline jspwiki-util/src/main/java/org/apache/wiki/util/TextUtil.java:236]: similar to above comment, perhaps is better to throw a `NullPointerException` informing which field should not be `null`?

**Review by juanpablo-santos — CHANGES_REQUESTED** [review]
changes requested

**Comment by arturobernalg** [comment]
@juanpablo-santos 
I agree with your comment. In fact, Joshua Bloch states: -->  Item 72 : Favor the use of standard exceptions:NullPointerException be thrown rather than IllegalArgumentException if caller passes null in some parameters for which null values are prohibited.
I clos the PR

**Pull request closed without merging**